repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityComparator.java

```diff
@@ -152,24 +156,49 @@ private AtlasEntityDiffResult getDiffResult(AtlasEntity updatedEntity, AtlasEnti
             }
         }
 
-        if (!skipClassificationCompare) {
+        if (context.isReplaceClassifications() || context.isReplaceTags() || context.isAppendTags()) {
             List<AtlasClassification> newVal  = updatedEntity.getClassifications();
             List<AtlasClassification> currVal = (storedEntity != null) ? storedEntity.getClassifications() : entityRetriever.getAllClassifications(storedVertex);
 
-            if (!Objects.equals(currVal, newVal)) {
-                diffEntity.setClassifications(newVal);
+            if (context.isReplaceClassifications()) {
+                if (!Objects.equals(currVal, newVal)) {
+                    diffEntity.setClassifications(newVal);
 
-                sectionsWithDiff++;
+                    sectionsWithDiff++;
 
-                if (findOnlyFirstDiff) {
-                    return new AtlasEntityDiffResult(diffEntity, true, false, false);
+                    if (findOnlyFirstDiff) {
+                        return new AtlasEntityDiffResult(diffEntity, true, false, false);
+                    }
+                }
+            } else {
+
+                Map<String, List<AtlasClassification>> diff;
+
+                if (context.isReplaceTags()) {
+                    diff = AtlasEntityUtils.getTagsDiffForReplace(updatedEntity.getGuid(),
```

Comment:
Can we reuse the function used in `setClassifications` API? this way `diff = ClassificationAssociator.Updater.computeChanges(new AtlasEntityHeader(storedEntity), new AtlasEntityHeader(updatedEntity));` after making the computeChanges static. Will keep the logic same

Reply:
As we are using same data structure `Map<String, List<AtlasClassification>>` to store the diff

Reply:
Or have common function for both in `EntityUtils`

Reply:
The difference between these 2 is that setClassifications does not maintain a list of unchanged tags.
I ensure the same notifications are being sent with all 3 flags (current relpaceClassifications sends all in-entity details) 
Also, AFAIK the flow in setClassifications filters out the propagated tags which will again differ the notifications sent in this flow, Want notifications to be the same as what we send with relaceClassifications flag